ConvivaIntegration/components/conviva/ConvivaAnalyticsTask.brs

```diff
@@ -147,30 +156,36 @@ sub onSourceUnloaded()
   m.sourceUnloadedTimer.mark() ' start the timer
 end sub
 
-sub onAdBreakStarted()
-  m.LivePass.detachStreamer()
-  m.LivePass.adStart()
-end sub
+function onAdBreakStarted()
+  m.livePass.detachStreamer()
+  m.livePass.adStart()
 
-sub onAdBreakFinished()
+  if m.adTrackingMode > 0 then m.adTracking.onAdBreakStarted()
+end function
+
+function onAdBreakFinished()
   m.LivePass.adEnd()
   m.LivePass.attachStreamer()
-end sub
+
+  if m.adTrackingMode > 0 then m.adTracking.onAdBreakFinished()
```

Comment:
As we have the `m.adTrackingMode > 0` check at least two times already, I suggest to extract it to a function with a meaningful name.

Reply:
That wouldn't make sense since all that function would do is to return `m.adTrackingMode`.
The idea behind this is to execute the code if a certain level of ad tracking is enable as well as if a higher level is enabled. 
The ad tracking mode levels have been extracted into an enum to make it more readable.